internal/start/start.go

```diff
@@ -158,14 +161,27 @@ func run(p utils.Program, ctx context.Context, fsys afero.Fs, excludedContainers
 		}); err != nil {
 			return errors.Errorf("failed to exec template: %w", err)
 		}
-		p.Send(utils.StatusMsg("Starting syslog driver..."))
+		p.Send(utils.StatusMsg("Starting vector..."))
+		var binds []string
+		env := []string{
+			"VECTOR_CONFIG=/etc/vector/vector.yaml",
+		}
+		// Special case for GitLab pipeline
+		host := utils.Docker.DaemonHost()
+		if parsed, err := client.ParseHostURL(host); err == nil {
+			if parsed.Scheme == "tcp" {
+				env = append(env, "DOCKER_HOST="+host)
+			} else {
+				// TODO: mount windows named pipe as unix domain socket?
+				// TODO: mount domain socket from docker rootless mode
+				binds = append(binds, "/var/run/docker.sock:/var/run/docker.sock:ro")
```

Comment:
Currently this relies on mounting root docker socket into container. Ideally we want to find a way to mount the user socket instead so it doesn't require root access.